bigquery/test/queries.test.js

```diff
@@ -145,62 +128,23 @@ describe('bigquery:query', function () {
     });
   });
 
-  describe('asyncPoll', function () {
+  describe('waitForJob', function () {
     it('should get the results of a job given its ID', function () {
       var example = getSample();
       var callback = sinon.stub();
       example.mocks.bigquery.job.returns(example.mocks.job);
 
-      example.program.asyncPoll(example.jobId, callback);
+      example.program.waitForJob(example.jobId, callback);
 
-      assert.equal(example.mocks.job.getMetadata.calledOnce, true);
-      assert.deepEqual(example.mocks.job.getMetadata.firstCall.args.slice(0, -1), []);
+      assert.equal(example.mocks.job.on.calledTwice, true);
+      assert.deepEqual(example.mocks.job.on.firstCall.args.slice(0, -1), ['error']);
```

Comment:
The sample is deterministic and the current assertions are about as restrictive/specific as it gets, and won't fail unless someone changes the sample. The only alternative would be looser assertions.